autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Protocol
+
+from autogen.agentchat import ConversableAgent
+
+from .document import Document
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, Protocol):
+    """
+    A graph rag agent is a conversable agent which could query graph database for answers.
+
+    An implementing agent class would
+    1. create a graph in the underlying database with input documents
+    2. use the retrieve() method to retrieve information.
+    3. use the retrieved information to generate and send back messages.
+
+    For example,
+    graph_rag_agent = GraphRagAgent(
+        name="graph_rag_agent",
+        max_consecutive_auto_reply=3,
+        retrieve_config={
+            "docs_path": [
+                "./data/movies.txt",
+            ],
+            "llm_config" = autogen.config_list_from_json("OAI_CONFIG_LIST")
```

Comment:
Thanks @randombet for getting this going, look forward to testing and using it.

Just wondering why there's an `llm_config` in the `retrieve_config` and not at the same level?

Reply:
I am reusing the OAI_CONFIG here. 
This LLM is used for graph db not for the agent (even though it could be the same)
For a few graph DB I investigated, they use LLM to build the graph, translate questions to DB queries and etc.